src/Microsoft.OData.Core/UriParser/SyntacticAst/ComputeToken.cs

```diff
@@ -0,0 +1,79 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ComputeToken.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+#if ODATA_CLIENT
+namespace Microsoft.OData.Client.ALinq.UriParser
+#else
+namespace Microsoft.OData.UriParser
+#endif
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Query token representing an Aggregate token.
+    /// </summary>
+    public sealed class ComputeToken : QueryToken
+    {
+        private QueryToken expression;
+
+        private string alias;
+
+        /// <summary>
+        /// Create an ComputeToken.
+        /// </summary>
+        /// <param name="expression">The computation token.</param>
+        /// <param name="alias">The alias for the computation.</param>
+        public ComputeToken(QueryToken expression, string alias)
+        {
+            ExceptionUtils.CheckArgumentNotNull(expression, "expression");
+            ExceptionUtils.CheckArgumentNotNull(alias, "alias");
+
+            this.expression = expression;
+            this.alias = alias;
+        }
+
+        /// <summary>
+        /// Gets the kind of this token.
+        /// </summary>
+        public override QueryTokenKind Kind
+        {
+            get { return QueryTokenKind.Compute; }

```

Comment:
format nit: the bracelets are on the line as the return, whereas they're on separate lines for the other properties below